pyfar/dsp/filter/fractional_octaves.py

```diff
@@ -509,3 +529,595 @@ def reconstructing_fractional_octave_bands(
+        >>> y_sum = pf.FrequencyData(np.sum(np.abs(y.freq)**2, 0),
+        ...     y.frequencies)
+        >>> pf.plot.freq(y)
+        >>> ax = pf.plot.freq(y_sum, color='k', log_prefix=10, linestyle='--')
+        >>> ax.set_title(
+        ...     "Filter bands and the sum of their squared magnitudes")
+        >>> plt.tight_layout()
+        >>> plt.show()
+
+    References
+    ----------
+    .. [#] International Electrotechnical Commission, International
+           Electrotechnical Commission und Technical Committee 29,
+           'Electroacoustics: Octave-Band and Fractional-Octave-Band
+           Filters.(IEC61260)', Part 1, Part 1, (2014),
+           ISBN: 978-2-8322-1417-6
+    """
+
+    def __init__(
+            self,
+            num_fractions=1,
+            frequency_range=(20.0, 20e3),
+            order=14,
+            sampling_rate=44100):
+
+        self._num_fractions = num_fractions
+        self._frequency_range = frequency_range
+        self._order = order
+        self._sampling_rate = sampling_rate
+
+        self._nominal_frequencies,\
+            self._center_frequencies,\
+            self._cutoff_frequencies = fractional_octave_frequencies(
+                self._num_fractions, self._frequency_range, True)
+
+        coefficients = self._get_coefficients(
+                sampling_rate=self._sampling_rate,
+                num_fractions=self._num_fractions,
+                frequency_range=self._frequency_range,
+                order=self._order)
+
+        # initialize superclass
+        super().__init__(coefficients, self._sampling_rate)
+        self.comment = (
+            f"Second order section 1/{num_fractions} fractional octave band"
+            "filter of order {order}")
+
+    def __repr__(self):
+        """Nice string representation of class instances"""
+        return (f"Energy-preserving fractional "
+                f"octave filter bank with {self.n_bands} "
+                f"bands between {self.frequency_range[0]} and "
+                f"{self.frequency_range[1]} Hz with "
+                f"{self.sampling_rate} Hz sampling rate and "
+                f"{self.num_fractions} per octave.")
+
+    def __eq__(self, other):
+        """Check for equality of two objects."""
+        return not DeepDiff(self.__dict__, other.__dict__)
+
+    @property
+    def num_fractions(self):
+        """Get the number of fractions per octave"""
+        return self._num_fractions
+
+    @property
+    def frequency_range(self):
+        """Get the frequency range of the filter bank in Hz"""
+        return self._frequency_range
+
+    @property
+    def nominal_frequencies(self):
+        """Get the IEC center frequencies of the (fractional)
+         octave bands in Hz"""
+        return self._nominal_frequencies
+
+    @property
+    def center_frequencies(self):
+        """Get the exact center frequencies of the (fractional)
+        octave bands in Hz"""
+        return self._center_frequencies
```

Comment:
The calculation can be done dynamically using the classes' properties.
```suggestion
        return f_crit = fractional_octave_frequencies(
            self.num_fractions, self.frequency_range, return_cutoff=True)[0]
```